packages/jazz-tools/src/tests/coMap.test.ts

```diff
@@ -1201,6 +1395,85 @@ describe("Creating and finding unique CoMaps", async () => {
     expect(foundAlice).toEqual(alice.id);
   });
 
+  test("manual upserting pattern", async () => {
+    // Schema
+    const Event = co.map({
+      title: z.string(),
+      identifier: z.string(),
+      external_id: z.string(),
+    });
+
+    // Data
+    const sourceData = {
+      title: "Test Event Title",
+      identifier: "test-event-identifier",
+      _id: "test-event-external-id",
+    };
+    const workspace = Group.create();
+
+    // Pattern
+    let eventId = Event.findUnique(
+      { identifier: sourceData.identifier },
+      workspace.id,
+    );
+    let activeEvent = await Event.load(eventId as unknown as ID<Event>);
+    if (!activeEvent) {
+      activeEvent = Event.create(
+        {
+          title: sourceData.title,
+          identifier: sourceData.identifier,
+          external_id: sourceData._id,
+        },
+        workspace,
+      );
+    } else {
+      activeEvent.applyDiff({
+        title: sourceData.title,
+        identifier: sourceData.identifier,
+        external_id: sourceData._id,
+      });
+    }
+    expect(activeEvent).toEqual({
+      title: sourceData.title,
+      identifier: sourceData.identifier,
+      external_id: sourceData._id,
+    });
+  });
+
+  test("upserting", async () => {
```

Comment:
We need additional tests for:
- upserting on an already available value
- upserting with a resolve on a new value
- upserting with a resolve on a new value where the passed refereces don't match the given resolve
- upserting with a resolve on an existing value
- upserting with a resolve on an existing value where the passed refereces don't match the given resolve